File: projectBlueWater/src/main/java/com/lasthopesoftware/storage/read/exceptions/StorageReadFileException.java
package com.lasthopesoftware.storage.read.exceptions;

import java.io.File;
import java.io.IOException;

public class StorageReadFileException extends IOException {
	private final File file;

	public StorageReadFileException(File file) {
		this(file, null);
	}

	public StorageReadFileException(File file, Exception innerException) {
		super("There was an error reading the serviceFile " + file + ".", innerException);

		this.file = file;
	}

	public File getFile() { return file; }
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/storage/read/exceptions/StorageReadFileException.kt
package com.lasthopesoftware.storage.read.exceptions

import java.io.File
import java.io.IOException

open class StorageReadFileException @JvmOverloads constructor(
    val file: File,
    innerException: Exception? = null
) : IOException(
    "There was an error reading the file $file.", innerException
)
